woocommerce/woocommerce-ios
# [Shipping Label] Add condition checks before showing "Email" and "Message" actions

Closes: #7983


## Description
If the customer address can't be verified during shipping label creation, the app will propose to contact the customer via email / message / call. Tapping on the "Email" or "Message" action does nothing if the device doesn't have Mail or Message apps configured.

This PR adds additional checks before displaying the "Email" and "Message" actions after user taps the "Contact Customer" button on the top banner.

## Testing instructions

Pre-requisite:
1. The test store must have WooCommerce Shipping & Tax plugin installed and activated.
2. Uninstall the Mail app on the device
3. Turn on the Airplane mode on the device, enable WiFi, and disable iMessage. This *should* prevent the app from being able to send out SMS.

Testing steps:
1. Create an order and collect the payment
4. Tap `Create Shipping Label` and fill in the `Ship from` data (first screen)
5. Fill the data for the `Ship to` screen with a semantically valid but non-existing address and tap `Done`
6. Once the `We were unable to verify shipping address...` alert is shown, tap the `Contact Customer` button
7. Notice the "Email" and "Messages" options will not be shown.

## Screenshots

<img src="https://user-images.githubusercontent.com/40906847/221174106-f9e29512-7f7d-440b-b7e8-0507520956ff.png" width=300/></td>

https://user-images.githubusercontent.com/40906847/221175007-df521b19-48ce-492b-884b-7b70f8e424ed.mp4


---
